packages/compiler-vapor/src/generate.ts

```diff
@@ -471,11 +480,16 @@ function genAppendNode(oper: AppendNodeIRNode, context: CodegenContext) {
   )
 }
 
-function genSetEvent(oper: SetEventIRNode, context: CodegenContext) {
+function genSetEvent(
+  oper: SetEventIRNode,
+  context: CodegenContext,
+  isInEffect: boolean,
+) {
   const { vaporHelper, push, newline, pushMulti, pushFnCall } = context
   const { keys, nonKeys, options } = oper.modifiers
 
   newline()
+  if (isInEffect) push('return ')
```

Comment:
If there are multiple operations in the single effect, it will early return

Reply:
I think removing effects should be handled in `on` function.

Reply:
Thank you for your input. I've revised the cleanup implementation to fulfill this requirement. Please review it again.